back/engines/commercial/insights/spec/acceptance/batch_assignments_spec.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rspec_api_documentation/dsl'
+
+resource 'Batch category assignments for view inputs' do
+  before { header 'Content-Type', 'application/json' }
+
+  let(:view) { create(:view) }
+  let(:view_id) { view.id }
+
+  let(:assignment_service) { Insights::CategoryAssignmentsService.new }
+  let(:json_response) { json_parse(response_body) }
+
+  shared_examples 'unauthorized requests' do
+    context 'when visitor' do
+      example 'unauthorized', document: false do
+        do_request
+        expect(status).to eq(401)
+      end
+    end
+
+    context 'when normal user' do
+      before { user_header_token }
+
+      example 'unauthorized', document: false do
+        do_request
+        expect(status).to eq(401)
+      end
+    end
+  end
+
+  shared_examples 'not-found requests' do
+    example "returns 404 if a category doesn't exist", document: false do
+      do_request(inputs: [], categories: ['bad-uuid'])
+      expect(status).to eq(404)
+    end
+
+    example "returns 404 if an input doesn't exist", document: false do
+      do_request(inputs: ['bad-uuid'], categories: [])
```

Comment:
Funny, bc I thought it would be clearer this way since in my mind "no category" cannot be missing :-) The other reason, I choose not put an actual value there is bc the test is define in a group of shared examples. So I'm trying to make minimal assumption about the context in which the group is injected.

What do you think about adding a third test with both inputs and category empty to make it clear that does not result in a 404?

Reply:
Still open to discussion here, but already merging it bc the next PR depends on it.

Reply:
No worries, thanks for taking the comments into account already, good improvements. When I approve a PR, I mean it, I'll always check for answers and improvements but you can merge anytime, especially if the changes made after the review are minimal. :slightly_smiling_face: